components/common/BraveCoreThemeProvider.tsx

```diff
@@ -0,0 +1,41 @@
+import * as React from 'react'
+import { ThemeProvider } from 'brave-ui/theme'
+import IBraveTheme from 'brave-ui/theme/theme-interface'
+
+export type BraveCoreThemeProviderProps = {
+  initialThemeType?: chrome.braveTheme.ThemeType
+  dark: IBraveTheme,
+  light: IBraveTheme
+}
+type BraveCoreThemeProviderState = {
+  braveCoreTheme?: chrome.braveTheme.ThemeType
+}
+
+export default class BraveCoreThemeProvider
+
+extends React.Component<BraveCoreThemeProviderProps, BraveCoreThemeProviderState> {
```

Comment:
nit: can we have everything in the same line? I usually define types for React classes as `Props` and `State` when there are no other classes in a file, in case a long line is not desirable

Reply:
Sure, I'm not sure about generic `Props` since that could be confusing with multiple components or when using intellisense or similar tools, but in this case it's not a problem at all. By the way, splitting the line on `<` and `>` seems to be a common practice so if we want lines less than 80-cols maybe we can agree on that in future cases?

Reply:
ya sounds good